src/internal/module/module.go

```diff
@@ -16,7 +21,32 @@ type Module struct {
 	TargetSystem string // The module target system
 }
 
+func (m Module) Logger(logger *slog.Logger) *slog.Logger {
+	return logger.With(slog.String("namespace", m.Namespace), slog.String("name", m.Name), slog.String("targetSystem", m.TargetSystem))
+}
```

Comment:
If we're going to have a method to enrich the logger we should explain what it does in the method name and also do it in a bit of a more maintainable way rather than just one line.

For example: https://github.com/opentofu/registry/blob/b148dfce0206c19baca80a1d4c137b97722067aa/src/lambda/api/moduleDownload.go#L24-L32

Also, if this is returning a logger, maybe something like `m.EnrichLogger(logger *slog.Logger)` would be good as that explains what it's doing.


If I didn't read the contents of this method, I would be confused why I would have to call: `logger := m.Logger(slog.Default())`, but I would understand why if it was something like `logger := m.EnrichLogger(slog.Default())`